src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -43,8 +44,8 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor)
             }
 
             var options = optionsAccessor.Value;
-            _entries = new Dictionary<object, CacheEntry>();
-            _entryLock = new ReaderWriterLockSlim();
+            _entries = new ConcurrentDictionary<object, CacheEntry>();
```

Comment:
In that case, I would use a private get-only property like 'EntriesAsCollection' which encapsulates the cast to improve readability, rather than adding another field.